src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -12500,6 +13390,363 @@ public void SynthesizedDelegateTypes_ParamsArray_DifferentElementTypes()
+
+            var source2 = """
+                public class D<T2> : C
+                {
+                    public override void M<T>(T[] xs) => System.Console.WriteLine("A");
+                    public void M(T2[] xs) => System.Console.WriteLine("B");
+                }
+                """;
+            var comp2 = CreateCompilation(source2, [comp1aRef]).VerifyDiagnostics();
+            var comp2Ref = comp2.EmitToImageReference();
+
+            var source1b = """
+                public abstract class C
+                {
+                    public abstract void M<T>(params T[] xs);
+                }
+                """;
+            var comp1b = CreateCompilation(source1b, assemblyName: "Lib1").VerifyDiagnostics();
+
+            var source3 = """
+                var c = ((C)new D<long>()).M<long>;
+                System.Console.WriteLine(c.GetType());
+                c(null);
+                var d = new D<long>().M;
+                System.Console.WriteLine(d.GetType());
+                d(null);
+                """;
+            var verifier = CompileAndVerify(source3, [comp2Ref, comp1b.EmitToImageReference()],
+                expectedOutput: """
+                <>f__AnonymousDelegate0`1[System.Int64]
+                A
+                System.Action`1[System.Int64[]]
+                B
+                """).VerifyDiagnostics();
+
+            var cm = verifier.Compilation.GetMember<IMethodSymbol>("C.M");
+            Assert.True(cm.Parameters.Single().IsParams);
+            var dm = verifier.Compilation.GetMembers<IMethodSymbol>("D.M").Single(m => m.OverriddenMethod is not null);
+            Assert.False(dm.Parameters.Single().IsParams);
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_ParamsArray_MostOverriddenMethod()
+        {
+            var source = """
+                var c = ((C)new D()).M;
+                System.Console.WriteLine(c.GetType());
+                var d = new D().M;
+                System.Console.WriteLine(d.GetType());
+
+                public class D : C
+                {
+                    public override void M(params int[] xs) { }
+                }
+
+                public abstract class C
+                {
+                    public abstract void M(int[] xs);
+                }
+                """;
+            var verifier = CompileAndVerify(source,
+                expectedOutput: """
+                System.Action`1[System.Int32[]]
+                System.Action`1[System.Int32[]]
+                """).VerifyDiagnostics();
+            var cm = verifier.Compilation.GetMember<IMethodSymbol>("C.M");
+            Assert.False(cm.Parameters.Single().IsParams);
+            var dm = verifier.Compilation.GetMember<IMethodSymbol>("D.M");
+            Assert.False(dm.Parameters.Single().IsParams);
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_ParamsArray_MostOverriddenMethod_Different()
```

Comment:
>SynthesizedDelegateTypes_ParamsArray_MostOverriddenMethod_Different [](http://example.com/codeflow?start=20&length=67)

The code that is setting up the test scenario is unnecessary complicated. We can start with `params` in abstract method and then drop `params` in source1b. Basically the code could be very similar to ```SynthesizedDelegateTypes_ParamsArray_LeastOverriddenMethod_Different``` only here we start with `params`, then drop in from abstract method.